services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/GenericS3TransferManager.java

```diff
@@ -457,9 +472,17 @@ public final Copy copy(CopyRequest copyRequest) {
 
         CompletableFuture<CompletedCopy> returnFuture = new CompletableFuture<>();
 
-        TransferProgressUpdater progressUpdater = new TransferProgressUpdater(copyRequest, null);
-        progressUpdater.transferInitiated();
-        progressUpdater.registerCompletion(returnFuture);
+        // set length to 10000 as reference value, since we don't make HeadObject call yet
+        TransferProgressUpdater progressUpdater = new TransferProgressUpdater(copyRequest, 10000L);
```

Comment:
Is this an accurate value ? The TransferProgressUpdater uses the total content size to calculate what fraction of the request body has been sent, etc. If this value is inaccurate, could lead to incorrect progress states being relayed to users.

Reply:
nvm found the answer below